packages/x-date-pickers/src/AdapterDateFns/AdapterDateFns.ts

```diff
@@ -85,13 +85,16 @@ export class AdapterDateFns
   implements MuiPickersAdapter<Date, DateFnsLocale>
 {
   constructor({ locale, formats }: AdapterOptions<DateFnsLocale, never> = {}) {
-    if (typeof addDays !== 'function') {
-      throw new Error(
-        [
-          'MUI: The `date-fns` package v3.x is not compatible with this adapter.',
-          'Please, install v2.x of the package or use the `AdapterDateFnsV3` instead.',
-        ].join('\n'),
-      );
+    /* istanbul ignore next */
+    if (process.env.NODE_ENV !== 'production') {
```

Comment:
I have reverted the previous changes and added this check to avoid bundling these warning in `production`.
Do you think it is worth it?

Reply:
Good for me

Reply:
For the `warnOnce`, it's definitely worth it